### PR TITLE
Added enhanced test-memory.php, originally from react/react

### PR DESCRIPTION
**Pull request by WyriHaximus**
Originally from: https://github.com/reactphp/react/blob/master/examples/test-memory.php

Example output:
screenshot from 2016-11-13 22 07 45

### CODE REVIEW

**Comment by clue** [comment]
Thanks, I think it makes sense to get this in :+1: The command arguments appear rather arbitrary to me, so I guess it may make sense to use options similar to https://github.com/reactphp/stream/pull/41?

We may also want to reconsider our defaults here. How about this?

```
php examples/benchmark-periodic-timers.php -t 5 -l StreamSelect
```

Also, while I understand why this is done, should we really default to adding a report every 2s?

**Comment by WyriHaximus** [comment]
@clue done, that is a great suggestion. Much cleaner this way :+1: .

When this is in I also want to take some examples from https://github.com/WyriHaximus/ReactBlogSeriesExamples and PR those as examples to this and other applicable repositories

**Comment by jsor** [comment]
What's the status here? Ready to merge?

**Comment by WyriHaximus** [comment]
FYI renamed the file to something better matching the currently included benchmarks

File: examples/95-benchmark-memory.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use React\EventLoop\LoopInterface;
+use React\EventLoop\Timer\TimerInterface;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$args = getopt('t:l:r:');
+$t  = isset($args['t']) ? (int)$args['t'] : 0;
+$loop = isset($args['l']) && class_exists('React\EventLoop\\' . $args['l'] . 'Loop') ? 'React\EventLoop\\' . $args['l'] . 'Loop' : Factory::create();
+
+if (!($loop instanceof LoopInterface)) {
```
Comment (kelunik) [inline examples/95-benchmark-memory.php]: Why that instead of just adding the `new` above?

**Pull request merged**